source/extensions/filters/http/ext_proc/http_client/http_client_impl.cc

```diff
@@ -110,18 +109,24 @@ void ExtProcHttpClient::onFailure(const Http::AsyncClient::Request&,
   ASSERT(reason == Http::AsyncClient::FailureReason::Reset ||
          reason == Http::AsyncClient::FailureReason::ExceedResponseBufferLimit);
   ENVOY_LOG(error, "Request failed: stream has been reset");
-  active_request_ = nullptr;
   onError();
 }
 
+const Envoy::StreamInfo::StreamInfo* ExtProcHttpClient::getStreamInfo() const {
+  if (active_request_ != nullptr) {
+    return &active_request_->streamInfo();
+  } else {
+    return nullptr;
+  }
+}
+
 void ExtProcHttpClient::onError() {
-  // Cancel if the request is active.
-  cancel();
   ENVOY_LOG(error, "ext_proc HTTP client error condition happens.");
   if (callbacks_) {
     callbacks_->onError();
     callbacks_ = nullptr;
   }
+  active_request_ = nullptr;
```

Comment:
Compared with previous code, it seems that you are missing active_request_->cancel();? 

why `cancel()` is removed from this function

Reply:
Please check comments above.